nix/flake/flakeref.go

```diff
@@ -103,10 +103,7 @@ func ParseRef(ref string) (Ref, error) {
 		parsed.Path = ref
 		return parsed, nil
 	}
-	parsed, fragment, err := parseURLRef(ref)
-	if fragment != "" {
-		return Ref{}, redact.Errorf("flake reference %q contains a URL fragment", ref)
-	}
```

Comment:
Take a look at `flake.Installable`, which I think does what you need. Flake references can't have attribute paths.

Reply:
The code that is deleted here was added as part of https://github.com/jetpack-io/devbox/commit/1dbcad274cddf480a6fab062dee2649a3b726112.  

It seems odd to throw an error when parsing a reference given that a fragment is a valid part of a flake url.  Leaving this code here causes

Reply:
Fragments aren't valid in a flake reference, but they are valid in an "installable". They're slightly different, but the distinction is important in some cases. For example:

```
$ nix eval --expr 'builtins.parseFlakeRef "nixpkgs#hello"'
error:
       … while calling the 'parseFlakeRef' builtin

         at «string»:1:1:

            1| builtins.parseFlakeRef "nixpkgs#hello"
             | ^

       error: unexpected fragment 'hello' in flake reference 'nixpkgs#hello'
```